src/arithmetic/arithmetic_expression.c

```diff
@@ -36,7 +36,7 @@ static AR_ExpNode *_AR_EXP_CloneOperand(AR_ExpNode *exp) {
 	switch(exp->operand.type) {
 	case AR_EXP_CONSTANT:
 		clone->operand.type = AR_EXP_CONSTANT;
-		clone->operand.constant = SI_CloneValue(exp->operand.constant);
+		clone->operand.constant = SI_ShallowCloneValue(exp->operand.constant);
```

Comment:
This is an annoying fix to a memory leak caused by the caching logic, which will replace const AST strings with heap-allocated strings that may not be freed properly. An example of this is the string 'const' in:
```
"MATCH (a)-[]->(b) WITH 'const' AS c MATCH (a), (b) RETURN a.val, b.val
```